File: 0153.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        if len(nums) is 1:
            return nums[0]
        
        k = len(nums)//2
        left = nums[0:k]
        right = nums[k:]
        
        return self.findMin(right) if not left or left[-1] > right[-1] else self.findMin(left)

File: snap_prep/0153.java
class Solution {
    public int findMin(int[] nums) {
        if(nums.length == 1){
            return nums[0];
        }
        
        int k = nums.length / 2;
        int[] left = Arrays.copyOfRange(nums, 0, k);
        int[] right = Arrays.copyOfRange(nums, k, nums.length);
        
        if(left.length == 0 || left[left.length-1] > right[right.length-1]){
            return findMin(right);
        }else{
            return findMin(left);
        }
    }
}
